### PR TITLE
Add some suggestions on how to handle flakes

### PR DESCRIPTION
**Pull request by spiffxp**
based on looking at deflaking PRs from the past few months,
what are some common anti-patterns or fixes?

### CODE REVIEW

File: contributors/devel/sig-testing/flaky-tests.md [hunk 1]
```diff
@@ -5,20 +5,58 @@ all tests are green, and we have a number of different CI systems running the
...
+  - Only ask for the resources you need
+    - https://github.com/kubernetes/kubernetes/pull/84975 - eg: drop memory constraints for test cases that only need cpu
+  - Don't use overly tight deadlines (but not overly broad either, non-[Slow] tests timeout after 5min)
+    - https://github.com/kubernetes/kubernetes/pull/85847 - eg: poll for `wait.ForeverTestTimeout` instead of 10s
+    - https://github.com/kubernetes/kubernetes/pull/84238 - eg: poll for 2min instead of 1min
+    - mark tests as [Slow] if they are unable to pass within 5min
+  - Do not expect actions to happen instantaneously or after a fixed delay
+  - Prefer informers and wait loops
+- Ensure the test provides sufficient context in logs for forensic debugging
+  - Explain what the test is doing
```
Comment (liggitt) [inline contributors/devel/sig-testing/flaky-tests.md]: Or why a polling loop is failing (example: "expected 3 widgets, found 2, will retry")

File: contributors/devel/sig-testing/flaky-tests.md [hunk 2]
```diff
@@ -5,20 +5,58 @@ all tests are green, and we have a number of different CI systems running the
...
+  - Explain what specific check failed
```
Comment (liggitt) [inline contributors/devel/sig-testing/flaky-tests.md]: Consider including details of the unexpected objects (example: "expected all items to be deleted, items foo, bar, baz still remain")

File: contributors/devel/sig-testing/flaky-tests.md [hunk 3]
```diff
@@ -5,20 +5,58 @@ all tests are green, and we have a number of different CI systems running the
...
+
+# Quarantining Flakes
+
+- Quarantine a single test case by adding `[Flaky]` to the test name in question,
+  most CI jobs exclude these tests. This makes the most sense for flakes that
+  are merge-blocking and taking too long to troubleshoot, or ocurring across
```
Comment (liggitt) [inline contributors/devel/sig-testing/flaky-tests.md]: occurring

File: contributors/devel/sig-testing/flaky-tests.md [hunk 4]
```diff
@@ -5,20 +5,58 @@ all tests are green, and we have a number of different CI systems running the
...
+  multiple jobs.
+  - eg: https://github.com/kubernetes/kubernetes/pull/83792
+  - eg: https://github.com/kubernetes/kubernetes/pull/86327
```
Comment (liggitt) [inline contributors/devel/sig-testing/flaky-tests.md]: When quarantining a presubmit test, ensure a frozen critical issue is present in the current milestone assigned to the owning sig for them to resolve the flakes and reintroduce the test, or determine the tested function is covered via another method and delete the test. Quarantined tests add very little value.

Comment (spiffxp) [reply contributors/devel/sig-testing/flaky-tests.md]: added as first bullet before describing how to quarantine

File: contributors/devel/sig-testing/flaky-tests.md (hunk 1 above)
Comment (spiffxp) [reply contributors/devel/sig-testing/flaky-tests.md]: added a few more examples for each of these

**Pull request merged**